invenio_files_rest/views.py

```diff
@@ -802,6 +831,47 @@ def delete(self, bucket=None, key=None, version_id=None, upload_id=None,
             obj = self.get_object(bucket, key, version_id)
             return self.delete_object(bucket, obj, version_id)
 
+
+class TagResource(ContentNegotiatedMethodView):
+    """Tag resource."""
+
+    def __init__(self, *args, **kwargs):
+        """Instantiate content negotiated view."""
+        super(TagResource, self).__init__(*args, **kwargs)
+
+    @pass_bucket
+    def put(self, key=None, bucket=None):
```

Comment:
I think we should add some sort of access rights check, i.e. `@need_bucket_permission('bucket-update')`.